syncing/src/main/kotlin/maru/syncing/ELSyncService.kt

```diff
@@ -41,17 +47,99 @@ data class ElBlockInfo(
  * status update callback `onStatusChange` and it tries to sync the EL by `executionLayerManager` with the latest known
  * EL hash from the `beaconChain`
  */
-class ELSyncServiceImpl(
+class ELSyncService(
   private val beaconChain: BeaconChain,
-  private val leeway: UInt,
   private val executionLayerManager: ExecutionLayerManager,
   private val onStatusChange: (ELSyncStatus) -> Unit,
+  private val config: Config,
+  private val timerFactory: (String, Boolean) -> Timer = { name, isDaemon -> Timer(name, isDaemon) },
 ) : LongRunningService {
+  data class Config(
+    val pollingInterval: Duration,
+  )
+
+  private val log = LogManager.getLogger(this.javaClass)
+
+  private var poller: Timer? = null
+  private var lastELSyncTarget: ElBlockInfo? = null
+  private var currentELSyncStatus: ELSyncStatus? = null
+
+  private fun pollTask() {
+    val latestBeaconBlockHeader = beaconChain.getLatestBeaconState().latestBeaconBlockHeader
+    val latestBeaconBlockNumber = latestBeaconBlockHeader.number
+    if (latestBeaconBlockNumber == 0UL) {
+      val newELSyncStatus = ELSyncStatus.SYNCED
+      if (currentELSyncStatus != newELSyncStatus) {
+        currentELSyncStatus = newELSyncStatus
+        onStatusChange(newELSyncStatus)
+      }
+      return
+    }
+
+    val latestSealedBeaconBlock =
+      beaconChain.getSealedBeaconBlock(
+        beaconBlockNumber = latestBeaconBlockHeader.number,
+      )!!
+    val newELSyncTarget =
+      ElBlockInfo(
+        blockNumber = latestSealedBeaconBlock.beaconBlock.beaconBlockBody.executionPayload.blockNumber,
```

Comment:
Nit: you could save some text to read and some pointers to follow by assignning the `latestSealedBeaconBlock.beaconBlock.beaconBlockBody.executionPayload` to a val